extensions/proposals/WEBGL_element_texture/extension.xml

```diff
@@ -0,0 +1,298 @@
+    an RGB value in the ITU-R Rec. 601 colorspace)</p>
+
+    <p>Sampling a WebGL texture which is not associated with any
+    <code>HTMLElementTexture</code> object will return a sample value of (0,0,0,1).</p>
+
+    <p>Each <code>TEXTURE_ELEMENT_WEBGL</code> texture object may require up
+    to 3 texture image units for each texture unit to which it is bound.
+    The number of texture image units required by a bound texture object
+    can be queried using <code>getTexParameter</code> with target set
+    to the texture target in question, value set to
+    <code>REQUIRED_TEXTURE_ELEMENT_IMAGE_UNITS_WEBGL</code>, and ActiveTexture
+    set to the texture unit to which the texture object is bound.</p>
+
+    <p><code>HTMLElementTexture</code> provides a commands for <em>latching</em> an
+    image frame into the consuming texture as its contents and retrieving additional
+    information as a timestamp or a transformation matrix.</p>
+
+    <p>This extension does not cover the details of how a <code>HTMLELementTexture</code>
+    object could be created from a producer element.
+    Different kinds of producers works differently and described in additional externsion
+    specifications which will extend this one.
+    Example of producer element specification:
+        WEBGL_video_texture</p>
+
+    <features>
+      <glsl extname="WEBGL_element_texture">
+        <stage type="vertex"/>
+
+        <stage type="fragment"/>
+
+        <type name="samplerElementWEBGL"/>
+
+        <function name="texture2D" type="vec4">
+          <param name="sampler" type="samplerElementWEBGL"/>
+
+          <param name="coord" type="vec2"/>
+        </function>
+
+        <function name="texture2DProj" type="vec4">
+          <param name="sampler" type="samplerElementWEBGL"/>
+
+          <param name="coord" type="vec3"/>
+        </function>
+
+        <function name="texture2DProj" type="vec4">
+          <param name="sampler" type="samplerElementWEBGL"/>
+
+          <param name="coord" type="vec4"/>
+        </function>
+
+      </glsl>
+    </features>
+  </overview>
+
+  <idl xml:space="preserve">
+[
+  NoInterfaceObject
+] interface HTMLElementTexture : EventTarget {
+  void updateTexImage();
+
+  double getTimestamp();
+  void getTransformMatrix(Float32Array matrix);
+
+  double getLastAvailableTimestamp();
+
+  attribute EventHandler onframeavailable;
+};
+
+[
+  NoInterfaceObject
+] interface WEBGL_element_texture {
+  const GLenum TEXTURE_ELEMENT_WEBGL = 0x9248;
+  const GLenum SAMPLER_ELEMENT_WEBGL = 0x9249;
+  const GLenum TEXTURE_BINDING_ELEMENT_WEBGL = 0x924A;
+  const GLenum REQUIRED_TEXTURE_ELEMENT_IMAGE_UNITS_WEBGL = 0x924B;
+};
+  </idl>
+
+  <!-- new functions -->
+
+  <newfun>
+    <p>On <code>HTMLElementTexture</code>:</p>
+
+    <function name="updateTexImage" type="void">
+      Update the texture image to the most recent frame from the image stream
+      and release the previously-held frame.
```

Comment:
Change the order. Put release first. E.g, "release the previously-held frame, if any, and update the texture image to the most recent frame."

Reply:
Unfortunately i forgot to add information how we should use bindTexture + updateTexImage().

I think updateTexImage() shouldn't latch the most recent frame from image stream if  our linked WebGL texture isn't bound to new texture target in active texture unit currently or need to use the SurfaceTexture way (it will implicitly bind its texture to the TEXTURE_VIDEO_WEBGL texture target.).